external/storm-jdbc/src/main/java/org/apache/storm/jdbc/common/JDBCClient.java

```diff
@@ -0,0 +1,209 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.jdbc.common;
+
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Lists;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.*;
+import java.sql.Date;
+import java.util.*;
+
+public class JDBCClient {
+    private static final Logger LOG = LoggerFactory.getLogger(JDBCClient.class);
+
+    private HikariDataSource dataSource;
+
+    public JDBCClient(Map<String, Object> map) {
+        Properties properties = new Properties();
+        properties.putAll(map);
+        HikariConfig config = new HikariConfig(properties);
+        this.dataSource = new HikariDataSource(config);
+    }
+
+    public int insert(String tableName, List<List<Column>> columnLists) {
+        Connection connection = null;
+        try {
+            connection = this.dataSource.getConnection();
+            StringBuilder sb = new StringBuilder();
+            sb.append("Insert into ").append(tableName).append(" (");
+            Collection<String> columnNames = Collections2.transform(columnLists.get(0), new Function<Column, String>() {
+                @Override
+                public String apply(Column input) {
+                    return input.getColumnName();
+                }
+            });
+            String columns = Joiner.on(",").join(columnNames);
+            sb.append(columns).append(") values ( ");
+
+            String placeHolders = StringUtils.chop(StringUtils.repeat("?,", columnNames.size()));
+            sb.append(placeHolders).append(")");
+
+            String query = sb.toString();
+            if(LOG.isDebugEnabled()) {
+                LOG.debug("Executing query " + query);
+            }
+
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            for(List<Column> columnList : columnLists) {
+                setPreparedStatementParams(preparedStatement, columnList);
+            }
+
+            return preparedStatement.executeUpdate();
```

Comment:
added a new config queryTimeOutSecs which is defaulted to 30 seconds which is equal to topology.message.timeout.secs. user can override during bolt construction.